### DISCUSSION TITLE
How to unbatch a big disconnected graph?

### DISCUSSION THREAD

**Discussion by BraveDistribution** [pyg-team/pytorch_geometric#4714] [Q&A]
Hello,

I am working with meshes and I have a DataBatch, with batch size of 4. I am interested in splitting the DataBatch into 4 objects (let's say the inverse of CollateFunction) in the data loader. 

Is there any native support? thanks.

**Comment by BraveDistribution** [comment]
https://github.com/pyg-team/pytorch_geometric/pull/4628/files/a83f0b005fb2a77ee243ac88aeeac4dbd5858050

unbatch functionality for tensor has been added, however the unbatch functionality for DataBatch is missing (fixing the changes in edge_indices).

**Reply by Padarn** [reply]
This seems like a reasonable thing to add. I'll create an issue out of this discussion. However, it would be fairly easy to manually adjust the edge_index manually for now.

Curious why you want to unbatch?

**Reply by BraveDistribution** [reply]
I use pytorch_geometric for generative task / motion prediction tasks. I need to visualize the generated graphs and compare it with the ground truth. They are 3d meshes. 

I use a function that was implemented, but not merged  https://github.com/pyg-team/pytorch_geometric/issues/1574 (they use tetras as additional parameter in graphs, which is not affected by the collater function, however this functionality might be useful for all the other graphs).

**Reply by Padarn** [reply]
thanks, I think its worth getting this merged, I opened a new issue here: https://github.com/pyg-team/pytorch_geometric/issues/4717

**Reply by rusty1s** [reply]
Isn't this already supported via `batch[i]` or `batch.to_data_list()`?

**Reply by Padarn** [reply]
(I hadn't looked into this, so please close the issue linked if this is the case @BraveDistribution @rusty1s )
